extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -2,114 +2,60 @@ package etcdsnapshot
 
 import (
 	"context"
-	"sort"
-	"strings"
+	"errors"
+	"fmt"
+	"net/url"
 	"time"
 
 	"github.com/rancher/norman/types"
 	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
 	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
 	"github.com/rancher/shepherd/clients/rancher"
 	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
 	rancherv1 "github.com/rancher/shepherd/clients/rancher/v1"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
 	"github.com/rancher/shepherd/extensions/clusters"
 	"github.com/rancher/shepherd/extensions/defaults"
-	"github.com/rancher/shepherd/extensions/defaults/stevetypes"
 	"github.com/sirupsen/logrus"
 	kwait "k8s.io/apimachinery/pkg/util/wait"
 )
 
 const (
 	ProvisioningSteveResouceType = "provisioning.cattle.io.cluster"
+	SnapshotSteveResourceType    = "rke.cattle.io.etcdsnapshot"
+	SnapshotClusterNameLabel     = "rke.cattle.io/cluster-name"
 	fleetNamespace               = "fleet-default"
 	localClusterName             = "local"
 	active                       = "active"
 	readyStatus                  = "Resource is ready"
+	oneSecond                    = 1000000000
 )
 
-// GetRKE1Snapshots is a helper function to get the existing snapshots for a downstream RKE1 cluster.
-func GetRKE1Snapshots(client *rancher.Client, clusterName string) ([]management.EtcdBackup, error) {
```

Comment:
Is this GetRKE1Snapshots definitely not used anywhere? Might be dangerous to remove the public function, but it's fine if we can confirm it's not being used at least in rancher/rancher.

Reply:
yes I confirmed; only was used in the tests I moved, and I replaced this functionality